src/main/java/core/mate/academy/model/TruckProducer.java

```diff
@@ -0,0 +1,12 @@
+package core.mate.academy.model;
+
+import core.mate.academy.service.MachineProducer;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {

```

Comment:
Parameterize `MachineProducer` in such a way that any implementation NOT using `Machine` or its subtype could NOT be created. Currently, the `TruckProducer` is correctly parameterized with `Truck`, which is a subtype of `Machine`, so this is good.